### PR TITLE
Added JOSS paper to submit

### PR DESCRIPTION
**Pull request by gaurav**
This PR adds the JOSS paper that we plan to submit for review. It also adds a "Community guidelines" section to [the README](https://github.com/phyloref/jphyloref/blob/c19b1788968b858529e699f60e276470556973e2/README.md) as recommended by JOSS (and because it's probably a good idea to include the link to our Github repository in our README anyway) as well as the [DOI to this repository archived in Zenodo](https://zenodo.org/badge/latestdoi/104808310).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,6 +23,12 @@ can be used to resolve [phyloreferences](http://phyloref.org):
   `input.owl` by comparing their resolution with the expected resolution recorded
   in the file.
 
+Full documentation of the API is included in the source code as [Javadoc] comments. It is also available online [at javadoc.io].
```
Comment (hlapp) [inline README.md:26]:
```suggestion
Documentation of the API is included in the source code as [Javadoc] comments. It is also available online [at javadoc.io].
```

File: paper/paper.bib [hunk 2]
```diff
@@ -0,0 +1,189 @@
+@UNPUBLISHED{Cellinese2021-gp,
```
Comment (hlapp) [inline paper/paper.bib:1]: I don't think `@UNPUBLISHED` is the appropriate type here? And can you please add the DOI and/or URL that uses the DOI.

File: paper/paper.bib [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+  publisher = "Springer Science and Business Media LLC",
+  volume    =  53,
+  number    =  1,
+  pages     = "1--61",
+  month     =  jun,
+  year      =  2014,
+  language  = "en"
+}
+
+@TECHREPORT{W3C_OWL_Working_Group2012-we,
```
Comment (hlapp) [inline paper/paper.bib:96]: Add URL.

**Review by hlapp — CHANGES_REQUESTED** [review]
Looks good overall. See inline comments for a few small issues.

File: paper/paper.bib (hunk 3 above)
Comment (gaurav) [reply paper/paper.bib:96]: Fixed in 6f0954a9d.

File: paper/paper.bib (hunk 2 above)
Comment (gaurav) [reply paper/paper.bib]: Fixed to type `article` in 22857eef489c. Also, I didn't notice that all the DOIs/URLs had vanished! I've replaced the DOIs in 699193cc238e70, and they now [appear in the rendered PDF](https://whedon.theoj.org/preview?id=d85a050de1a01dfc9a8176d6).

**Review by hlapp — APPROVED** [review]
👍🏻🎉

**Pull request merged**